eras/shelley/impl/src/Cardano/Ledger/Shelley/RewardUpdate.hs

```diff
@@ -70,25 +70,30 @@ import NoThunks.Class (NoThunks (..), allNoThunks)
 
 -- ===============================================================
 
+type RewardEvent c = (Map (Credential 'Staking c) (Set (Reward c)))
+
 -- | The result of reward calculation is a pair of aggregate Maps.
-newtype RewardAns c
-  = RewardAns (Map (Credential 'Staking c) (Reward c))
+--   One for the accumulated answer, and one for the answer since the last pulse
+data RewardAns c = RewardAns
+  { accumRewardAns :: !(Map (Credential 'Staking c) (Reward c)),
+    recentRewardAns :: !(RewardEvent c)
+  }
   deriving (Show, Eq, Generic)
   deriving (NFData)
 
 instance NoThunks (RewardAns crypto)
 
 instance CC.Crypto c => ToCBOR (RewardAns c) where
-  toCBOR (RewardAns x) = toCBOR x
+  toCBOR (RewardAns accum recent) = encodeListLen 2 <> toCBOR accum <> toCBOR recent
 
 instance CC.Crypto c => FromCBOR (RewardAns c) where
-  fromCBOR = RewardAns <$> fromCBOR
+  fromCBOR = decodeRecordNamed "RewardAns" (const 2) (RewardAns <$> fromCBOR <*> fromCBOR)
 
 -- | The provenance we collect
 type KeyHashPoolProvenance c = Map (KeyHash 'StakePool c) (RewardProvenancePool c)
 
 -- | The type of RewardPulser we pulse on.
-type Pulser c = RewardPulser c (ProvM (KeyHashPoolProvenance c) ShelleyBase) (RewardAns c)
+type Pulser c = RewardPulser c ShelleyBase (RewardAns c)
```

Comment:
thank you for cleaning this up!

Reply:
Still one more piece of provenence we can remove.